docs/Using-Primary-Outputs-for-Post-Actions.md

```diff
@@ -0,0 +1,408 @@
+      "path": "MyTestProject.csproj"        
+    }
+  ],
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+      "args": {
+        "files": [ "./Custom/Path/MyTestProject.csproj" ]
+      }
+    }
+  ]
+}
+```
+Note that the example above contains primary output with original name in the final location. The actual primary output returned by API, will contain the final file name after renames are applied.
+Post action `files` should contain the original location in template definition.
+Example:
+```
+dotnet new TestAssets.TemplateWithSourceNameAndCustomSourcePath --name Awesome 
+```
+The primary output returned by API is: `Awesome.csproj`.
+The final location of the files is: 
+- `./Awesome.csproj`
+- `./Class.cs`
+
+### Changing target
+When changing the target for template instantiation, make sure that primary output path contains final location of the file after instantiation including target path.
+Example:
+template files:
+- `MyTestProject.csproj`
+- `Class.cs`
+
+`template.json` definition
+```
+{
+  "author": "Test Asset",
+  "classifications": [ "Test Asset" ],
+  "name": "TemplateWithSourceNameAndCustomTargetPath",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.TemplateWithSourceNameAndCustomTargetPath",
+  "precedence": "100",
+  "identity": "TestAssets.TemplateWithSourceNameAndCustomTargetPath",
+  "shortName": "TestAssets.TemplateWithSourceNameAndCustomTargetPath",
+  "sourceName": "MyTestProject",
+  "sources": [
+    {
+      "target": "./Custom/Path/"
+    }
+  ],
+  "primaryOutputs": [
+    {
+      "path": "./Custom/Path/MyTestProject.csproj"        
+    }
+  ],
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+      "args": {
+        "files": [ "MyTestProject.csproj" ]
+      }
+    }
+  ]
+}
+```
+Note that the example above contains primary output with original name in the final location. The actual primary output returned by API, will contain the final file name after renames are applied.
+Post action `files` should contain the original location in template definition.
+Example:
+```
+dotnet new TestAssets.TemplateWithSourceNameAndCustomTargetPath --name Awesome 
+```
+The primary output returned by API is: `/Custom/Path/Awesome.csproj`.
+The final location of the files is: 
+- `./Custom/Path/Awesome.csproj`
+- `./Custom/Path/Class.cs`
+
+### Changing source and target
+When changing both source and target for template instantiation, make sure that primary output path contains final location of the file after instantiation including target path.
```

Comment:
Should we add to the end:
```
without the "./" at the beginning.
```

Otherwise If I read this line, I would write the following as my primary output:
```
./Target/Output/MyTestProject.csproj
```

Reply:
in fact it is a mistake :) we encourage people to use `./` instead of skipping it.